extensions/data-transfer/portability-data-transfer-koofr/src/main/java/org/datatransferproject/transfer/koofr/photos/KoofrPhotosExporter.java

```diff
@@ -0,0 +1,163 @@
+import org.datatransferproject.types.transfer.auth.TokensAndUrlAuthData;
+
+public class KoofrPhotosExporter
+    implements Exporter<TokensAndUrlAuthData, PhotosContainerResource> {
+
+  private final Monitor monitor;
+
+  private final KoofrClientFactory koofrClientFactory;
+
+  public KoofrPhotosExporter(KoofrClientFactory koofrClientFactory, Monitor monitor) {
+    this.koofrClientFactory = koofrClientFactory;
+    this.monitor = monitor;
+  }
+
+  @Override
+  public ExportResult<PhotosContainerResource> export(
+      UUID jobId, TokensAndUrlAuthData authData, Optional<ExportInformation> exportInformation)
+      throws CopyExceptionWithFailureReason {
+    Preconditions.checkNotNull(authData);
+
+    KoofrClient koofrClient = koofrClientFactory.create(authData);
+
+    String rootPath = koofrClient.getRootPath();
+
+    List<FilesListRecursiveItem> items;
+
+    try {
+      items = koofrClient.listRecursive(rootPath);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+
+    ArrayList<PhotoAlbum> albums = new ArrayList<>();
+    HashSet<String> albumsWithPhotos = new HashSet<>();
+
+    ArrayList<PhotoModel> exportPhotos = new ArrayList<>();
+
+    for (FilesListRecursiveItem item : items) {
+      String path = item.getPath();
+      String[] pathParts = path.split("/");
+
+      if (FilesListRecursiveItem.TYPE_FILE.equals(item.getType())) {
+        FilesFile file = item.getFile();
+
+        if (FilesFile.TYPE_DIR.equals(file.getType())) {
+          if (pathParts.length == 2 && !pathParts[1].isEmpty()) {
+            String albumId = path;
+
+            String albumName = file.getName();
+
+            if (file.getTags() != null && file.getTags().containsKey("originalName")) {
+              albumName = file.getTags().get("originalName").get(0);
+            }
+
+            String description = null;
+
+            if (file.getTags() != null && file.getTags().containsKey("description")) {
+              description = file.getTags().get("description").get(0);
+            }
+
+            albums.add(new PhotoAlbum(albumId, albumName, description));
+          }
+        } else if (pathParts.length > 2 && file.getContentType().startsWith("image/")) {
+          String photoId = path;
+
+          String photoName = file.getName();
+
+          String description = null;
+
+          if (file.getTags() != null && file.getTags().containsKey("description")) {
+            description = file.getTags().get("description").get(0);
+          }
+
+          String contentType = file.getContentType();
+
+          String albumId = String.join("/", Arrays.copyOfRange(pathParts, 0, 2));
+          albumsWithPhotos.add(albumId);
+
+          String fullPath = rootPath + path;
+
+          String fetchableUrl;
+
+          try {
+            fetchableUrl = koofrClient.fileLink(fullPath);
+          } catch (IOException e) {
+            monitor.severe(() -> String.format("Koofr file link error: %s: %s", path, e));
```

Comment:
The monitor can handle the exception separately and it can format nicer if passed as a separate param.

```suggestion
            monitor.severe(() -> String.format("Koofr file link error: %s", path), e));
```